src/main/scala/com/emarsys/jwt/akka/http/JwtAuthentication.scala

```diff
@@ -25,6 +25,10 @@ trait JwtAuthentication {
 
   def jwtAuthenticate[UserData](um: FromStringUnmarshaller[UserData]): Directive1[UserData] = for {
     jwtToken <- optionalHeaderValueByName("Authorization").map(stripBearerPrefix)
+    userData <- jwtAuthenticateFromOptionString(jwtToken, um)
+  } yield userData
+
+  def jwtAuthenticateFromOptionString[UserData](jwtToken: Option[String], um: FromStringUnmarshaller[UserData]): Directive1[UserData] = for {
```

Comment:
I think the following name would be an little improvement.

```suggestion
  def jwtAuthenticateToken[UserData](jwtToken: Option[String], um: FromStringUnmarshaller[UserData]): Directive1[UserData] = for {
```

Reply:
ok

Reply:
amended

Reply:
Thanks